Optimize GroupNormOp

xiaomengy opened the pull request:
Summary: Optimize GroupNormOp

Differential Revision: D10455567

BIT-silence commented:
Some benchmark

On CPU, N = 128, C = 128, H = 64, W = 64, group = 32:
NCHW 214.66ms -> 152.50ms
NHWC 912.79ms -> 848.54ms

On GPU, N = 256, C = 256, H = 128, W = 128, group = 32:
NCHW 134.58ms -> 70.74ms
NHWC 221.92ms -> 102.61ms

houseroad reviewed (approved):
LG, the failures are not related. Great improvement.

xiaomengy committed:
Optimize GroupNormOp (#12844)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/12844

Optimize GroupNormOp

Reviewed By: houseroad

Differential Revision: D10455567

fbshipit-source-id: 40d075fb1e2fc01bc4c2a32e871a38c1ff9c91ef
